### PR TITLE
refactor: migrate codebase to foundry-template

### LINKED ISSUE #3: Migrate to foundry-template
Since we're planning to maintain this repository, we need to move it to our foundry-template to ensure it comes with the same workflows.

This means moving the template over, porting the tests, and ensuring CI is happy.

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
Closes #3

### CODE REVIEW

**Comment by 3esmit** [comment]
I guess we need to first update contracts?

**Comment by 3esmit** [comment]
btw, dont touch the contracts in this PR, because we are going to have conflicts with #1

File: README.md
Comment (3esmit) [inline README.md:1]: Don't replace the README.md

**Comment by 0x-r4bbit** [comment]
@3esmit that's correct. I have not touched the contracts in this PR, this is merely migrating to our foundry template, but excluding porting the tests. For the test ports I need #1 so I've prepared a separate PR in #9 for that

**Review by 3esmit — CHANGES_REQUESTED** [review]
Just don't replace README.md

File: readme-header.png
Comment (3esmit) [inline readme-header.png:1]: don't delete this

**Comment by 0x-r4bbit** [comment]
Updated according to your comments.

**Review by 3esmit — APPROVED** [review]

**Pull request merged**